conda_index/index/__init__.py

```diff
@@ -540,74 +547,79 @@ def extract_wrapper(args: tuple):
 
                 # Collect repodata from packages, save to
                 # REPODATA_FROM_PKGS_JSON_FN file
-                t = tqdm(
-                    extract_subdirs_to_cache(),
-                    total=len(subdirs),
-                    disable=(verbose or not progress),
-                    leave=False,
-                )
-                for subdir in t:
-                    t.set_description("Subdir: %s" % subdir)
-                    t.update()
-                    with tqdm(
-                        total=8, disable=(verbose or not progress), leave=False
-                    ) as t2:
-                        t2.set_description("Gathering repodata")
-                        t2.update()
-                        log.debug("gather repodata")
-
-                        repodata_from_packages = self.index_subdir(
-                            subdir, verbose=verbose, progress=progress
+                with self.thread_executor_factory() as index_process:
+                    futures = [
+                        index_process.submit(
+                            functools.partial(
+                                self.index_prepared_subdir,
+                                subdir=subdir,
+                                verbose=verbose,
+                                progress=progress,
+                                patch_generator=patch_generator,
+                                current_index_versions=current_index_versions,
+                            )
                         )
+                        for subdir in extract_subdirs_to_cache()
+                    ]
+                    for f in futures:
```

Comment:
clean up before merge

Reply:
I thought the with statement: would wait for all the submitted futures, but it appears we have to call .result() on each.

Reply:
Not crazy about having important side effects happen inside logging statements.

```suggestion
                    for f in futures:
                        result = f.result()
                        log.debug(f"From subprocess {result=}")
```